azure-iot-device/azure/iot/device/provisioning/provisioning_device_client.py

```diff
@@ -12,31 +12,35 @@
 from azure.iot.device.common.evented_callback import EventedCallback
 from .abstract_provisioning_device_client import AbstractProvisioningDeviceClient
 from .abstract_provisioning_device_client import log_on_register_complete
-from .internal.polling_machine import PollingMachine
+from azure.iot.device.provisioning.pipeline import constant as dps_constant
+from .pipeline import exceptions as pipeline_exceptions
+from azure.iot.device import exceptions
+
 
 logger = logging.getLogger(__name__)
 
 
+def wait_for_completion(callback):
```

Comment:
lets call this `handle_result` for parity with the iothub client, and to prevent confusion with the `.wait_for_completion()` method